app/controllers/register_controller.rb

```diff
@@ -33,21 +34,30 @@ def edit
   end
 
   def confirm
-    return true if params[:data_confirmed]
-    @register = get_register(params[:register])
-
-    @current_register_record = OpenRegister.record(params[:register].downcase,
-                        params[params[:register].downcase.to_sym],
-                        :beta)
-
-    if @current_register_record != nil
-      @current_register_record = convert_register_json(@current_register_record)
+    errors = @data_validator.get_form_errors(params,@registers_client.get_register(params[:register], 'beta').get_field_definitions)
```

Comment:
* Needs spaces after `,`
* It may be better to build a model with validations (ActiveModel) then ask it if it's valid -- that way the errors that come back will tell you exactly what's wrong with no extra code in `lib`
* Long line
* Might be a good idea to break up and use some intermediate vars